src/scripts/modules/add_destination.js

```diff
@@ -0,0 +1,105 @@
+import destinationsList from "../destinationList.js";
+import events from "./pubsub.js";
+
+let form;
+let radioBtns;
+let radioVisited;
+let radioBucketlist;
+let cityInput;
+let countryInput;
+let photoInput;
+let storedCity;
+//new list item state before the item is added
+let formState = {
+  city: null,
+  country: null,
+  photo: null,
+};
+
+function cacheDom() {
+  form = document.getElementById("add-city-form");
+  uploadBtn = document.getElementById("upload");
+  radioBtns = document.querySelectorAll("input[type=radio]");
+  radioVisited = document.getElementById("rd-visited");
+  radioBucketlist = document.getElementById("rd-bucketlist");
+  cityInput = document.getElementById("inputCity");
+  countryInput = document.getElementById("inputCountry");
+  photoInput = document.getElementById("inputImg");
+}
+
+function bindEvents() {
+  form.addEventListener("submit", (event) => {
+    addDestination();
+    event.preventDefault();
+    //get the index of the destination pushed to the destinationsList and publish it
+    const itemIndex = destinationsList.length - 1;
+    events.publish("destinationAdded", itemIndex);
+    storeNew();
+    form.reset();
+  });
+  document.addEventListener("DOMContentLoaded", loadNew);
+  radioVisited.addEventListener("click", isVisited);
+  radioBucketlist.addEventListener("click", isBucketlist);
+  cityInput.addEventListener("change", (event) => {
+    updateFormState("city", event.target.value);
+  });
+  countryInput.addEventListener("change", (event) => {
+    updateFormState("country", event.target.value);
+  });
+  photoInput.addEventListener("change", (event) => {
+    updateFormState("photo", event.target.value);
+  });
+}
+
+//update the formState values for city, country and photo on 'change' event
+function updateFormState(fieldName, value) {
+  formState[fieldName] = value;
+}
+
+function storeNew() {
+  storedCity = localStorage.setItem(
+    "newDestination",
+    JSON.stringify(formState)
+  );
+}
+
+// localStorage.clear();
+
+function loadNew() {
+  let isStored = localStorage.getItem("newDestination");
+  if (isStored) {
+    let data = JSON.parse(isStored);
+  } else {
+    data = [];
+  }
+  localStorage.setItem("newDestination", data);
+}
```

Comment:
I see what you are trying to do here, it's almost there. Maybe comment it out for now and continue in a next PR :)